src/CheckoutSdk/Payments/CustomerSource.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+
+namespace Checkout.Sdk.Payments
+{
+    public class CustomerSource : IPaymentSource
+    {
+        public CustomerSource(string id, string email)
+        {
+            if (string.IsNullOrWhiteSpace(id)
+                && string.IsNullOrWhiteSpace(email))
+            {
+                throw new CheckoutException($"Either {id} or {email} must be provided.");
+            }
+
+            if (!string.IsNullOrWhiteSpace(id))
+            {
+                Id = id;
+            }
+
+            if (!string.IsNullOrWhiteSpace(email))
+            {
+                if(email?.Split('@').Length != 2)
+                {
+                    throw new CheckoutException($"{email} contain one @");
```

Comment:
Throw System.FormatException. That said, I'd be tempted to let the API handle this sort of validation (ask my why tomorrow)